packages/patternfly-3/patternfly-react-extensions/src/components/Select/Select.js

```diff
@@ -0,0 +1,176 @@
+          <EllipsisWithTooltip>{selectedItem.name}</EllipsisWithTooltip>
+        ) : (
+          <EllipsisWithTooltip>
+            <span className="select-toggle-placeholder">{placeholder}</span>
+          </EllipsisWithTooltip>
+        )}
+        <span className="caret select-caret" />
+      </Button>
+      {open && (
+        <div className="select-body-container">
+          {isSearchable && (
+            <SelectInput
+              onClear={onSearchClear}
+              focus={focusOnMount}
+              onChange={onSearchChange}
+              onKeyDown={onKeyDown}
+              placeholder={placeholder}
+              searchValue={searchValue}
+            />
+          )}
+          <ListGroup className="select-scrollable-list">
+            {isLoading ? (
+              <ListGroupItem id="select-loading" key="loading" className="select-loading-list">
+                <div id="select-loading-container">
+                  <Spinner id="select-spinner" loading size="sm" /> <span>{loadingText}</span>
+                </div>
+              </ListGroupItem>
+            ) : (
+              renderResults(options)
+            )}
+          </ListGroup>
+        </div>
+      )}
+    </div>
+  );
+};
+
+Select.propTypes = {
+  /** cursor position for keyboard navigation */
+  cursor: PropTypes.number,
+
+  /** is Select disabled */
+  disabled: PropTypes.bool,
+
+  /** no Results text */
+  emptyText: PropTypes.string,
+
+  /** should Search input take focus on open */
+  focusOnMount: PropTypes.bool,
+
+  /** isLoading bool */
+  isLoading: PropTypes.bool,
+
+  /** show Search bool */
+  isSearchable: PropTypes.bool,
+
+  /** isLoading text */
+  loadingText: PropTypes.string,
+
+  /** onItemClick func({ event, id, name }) */
+  onItemClick: PropTypes.func,
+
+  /** onKeyDown func(event) */
+  onKeyDown: PropTypes.func,
+
+  /** onSearchChange func(event) */
+  onSearchChange: PropTypes.func,
+
+  /** onSearchClear func() */
+  onSearchClear: PropTypes.func,
+
+  /** onToggle func() */
+  onToggle: PropTypes.func,
+
+  /** isOpen bool */
+  open: PropTypes.bool.isRequired,
+
+  /** options array */
+  options: PropTypes.arrayOf(
+    PropTypes.shape({
+      id: PropTypes.string,
+      name: PropTypes.string
+    })
+  ).isRequired,
+
+  /** placeholder */
+  placeholder: PropTypes.string,
+
+  /** searchValue */
+  searchValue: PropTypes.string,
+
+  /** selectedItem object { id: string, name: string } */
+  selectedItem: PropTypes.PropTypes.shape({
+    id: PropTypes.string,
+    name: PropTypes.string
+  })
+};
+
+Select.defaultProps = {
+  disabled: false,
+  cursor: -1,
+  emptyText: 'No Results',
+  focusOnMount: true,
+  isLoading: false,
+  isSearchable: true,
+  loadingText: 'Loading...',
+  onItemClick: noop,
+  onKeyDown: noop,
+  onSearchChange: noop,
+  onSearchClear: noop,
+  onToggle: noop,
+  placeholder: 'Filter...',
+  searchValue: '',
+  selectedItem: {
+    id: '10',
+    name: 'ten'
+  }
```

Comment:
could cause issues for applications not supplying a selectedItem, if they happen to have a matching item it will be inadvertently selected.